gdal/ogr/ogrsf_frmts/flatgeobuf/ogr_flatgeobuf.h

```diff
@@ -144,19 +154,43 @@ class OGRFlatGeobufLayer final : public OGRLayer
         void VerifyBuffers( int bFlag ) { m_bVerifyBuffers = CPL_TO_BOOL(bFlag); }
 
         const std::string& GetFilename() const { return m_osFilename; }
+
+        static std::string GetTempFilePath(CPLString fileName, char **papszOptions);
+
+        uint16_t GetIndexNodeSize() const { return m_indexNodeSize; } 
+        OGRwkbGeometryType GetOGRGType() const { return m_eGType; } 
+};
+
+class OGRFlatGeobufEditableLayer final: public OGREditableLayer
+{
+    std::set<CPLString> m_oSetFields;
+
+    OGRFlatGeobufLayer *m_poFlatGeobufLayer;
```

Comment:
should likely be a std::unique_ptr<OGRFlatGeobufLayer> so it gets destroyed

Reply:
I'm not sure.. the pointer is passed to the OGREditableLayer constructor which appears to take ownership?

Reply:
I was able to remove it completely and instead use the protected member from the base class.